arangod/V8Server/v8-user-functions.cpp

```diff
@@ -0,0 +1,153 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2016 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Wilfried Goesgens
+////////////////////////////////////////////////////////////////////////////////
+
+#include "v8-user-functions.h"
+
+//#include "Basics/Result.h"
+
+#include "v8-vocbaseprivate.h"
+#include <v8.h>
+#include "V8/v8-utils.h"
+#include "V8/v8-vpack.h"
+#include "V8/v8-conv.h"
+
+#include "VocBase/Methods/AqlUserFunctions.h"
+
+using namespace arangodb;
+
+static void JS_UnregisterAQLUserFunction(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("UNREGISTER_AQL_USER_FUNCTION(<name>)");
+  }
+
+  TRI_vocbase_t* vocbase = GetContextVocBase(isolate);
+
+  std::string functionName = TRI_ObjectToString(args[0]);
+
+  Result rv = unregisterUserFunction(vocbase, functionName);
+
+  if (rv.fail()) {
+    TRI_V8_THROW_EXCEPTION(rv);
+  }
```

Comment:
does this function have any return value in case of success? should it?

Reply:
it doesn't. However, the api will return 1 or throw an error, I'll handle that similary here.

Reply:
no need to modify it if we do not need a return value here. was just wondering.